libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -361,29 +340,27 @@ fn handle_request(req: CmRequest, state: &mut ConnectionState) {
 }
 
 fn notify_subscribers(
-    subscribers: &mut HashMap<String, SyncSender<CmNotification>>,
+    subscribers: &mut Vec<SyncSender<CmNotification>>,
     notification: CmNotification,
 ) {
-    for (id, sender) in subscribers.clone() {
-        if let Err(_) = sender.send(notification.clone()) {
+    subscribers.retain(|sender| {
+        if sender.send(notification.clone()).is_err() {
             warn!("subscriber has dropped its connection to connection manager");
```

Comment:
Do we need some sort of id for this to be useful? How would we use this message when problem solving?

Reply:
Hmm maybe this shouldn't be `warn!` since it's completely valid for the subscribe to drop out of scope.